src/resolvers.ts

```diff
@@ -23,25 +21,51 @@ type LoginInfo = {
 
 export const resolvers = {
   Query: {
-    user: async (_, { id }: { id: number }, ctx: UserInterface) => {
-      if (ctx === null) {
-        throw new CustomError("Invalid Token", 401);
+    user: async (_, { id }: { id: number }, ctx: GraphQLContext) => {
+      if (!ctx || !ctx.id) {
+        throw new CustomError("Unauthenticated", 401);
       }
       const dbUser = await UserRepository.findOne({ where: { id } });
       if (!dbUser) {
         throw new CustomError("User not found", 400);
       }
       return dbUser;
     },
+
+    users: async (
+      _,
+      { amount = 10 }: { amount?: number },
+      ctx?: GraphQLContext
+    ) => {
+      if (!ctx || !ctx.id) {
+        throw new CustomError("Unauthenticated", 401);
+      }
+      if (amount <= 0) {
+        throw new CustomError("Amount of users must be greater than 0", 400);
+      }
+
+      const userArray: UserPayload[] = await UserRepository.find({
```

Comment:
o tipo já indica que é uma array, fica meio redundante. O plural funciona bem
```suggestion
      const users: UserPayload[] = await UserRepository.find({
```